### ISSUE TITLE
TFL Detect "TFLite Demo has stopped." error using provided pets sample assets

### ISSUE THREAD

**Issue by blakefox** [tensorflow/tensorflow#27791]
**System information**
- Linux Ubuntu 16.04 and / or ( Mint 18 ):
- TensorFlow installed from binary:
- TensorFlow version 1.12.0 ( Compiled without AVX ):
- Python: 3.5.2 
- Cuda: 9.0
- Cudnn: 7.4
- GPU: 1050ti
- System: Dual  Xeon E5540 with 56GB Memory

I am having an unique issue implementing a custom trained model into the Android Demo located here within the tensorflow source

[https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/examples/android/app](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/examples/android/app)

specifically, i followed the tutorial located at

https://medium.com/tensorflow/training-and-serving-a-realtime-mobile-object-detector-in-30-minutes-with-cloud-tpus-b78971cf1193

after Android Studio is installed and configured, I clone a clean version of the tensorflow source, set up the Workspace and then run

`bazel build -c opt --config=android_arm{,64} --cxxopt='--std=c++11' tensorflow/lite/examples/android:tflite_demo`

The build succeeds with a couple warnings, when pushed to my device ( Android Note 8 ) via

`adb install -r tflite_demo.apk`

All three apps work great!.

The problem starts when I try to utilize the sample model and label file linked in the tensorflow tutorial on medium.com

[https://storage.googleapis.com/download.tensorflow.org/models/tflite/pets_ssd_mobilenet_v1_0.75_quant_2018_06_29.zip](https://storage.googleapis.com/download.tensorflow.org/models/tflite/pets_ssd_mobilenet_v1_0.75_quant_2018_06_29.zip)

i place both the detect.tflite and the pets_labels_list.txt in the following location

`//tensorflow/lite/examples/android/app/src/main/assets`

and change the following source files as instructed with the following changes

BUILD file -> //tensorflow/lite/examples/android

`
assets = [        "//tensorflow/lite/examples/android/app/src/main/assets:labels_mobilenet_quant_v1_224.txt",
        "@tflite_mobilenet_quant//:mobilenet_v1_1.0_224_quant.tflite",
        "@tflite_conv_actions_frozen//:conv_actions_frozen.tflite",
        "//tensorflow/lite/examples/android/app/src/main/assets:conv_actions_labels.txt",
        "@tflite_mobilenet_ssd//:mobilenet_ssd.tflite",
        "//tensorflow/lite/examples/android/app/src/main/assets:detect.tflite",
        "//tensorflow/lite/examples/android/app/src/main/assets:box_priors.txt",
        "//tensorflow/lite/examples/android/app/src/main/assets:labels.txt",
    ],
`

DetectorActivity.java -> //tensorflow/lite/examples/android/app/src/main/java/org/tensorflow/demo

`
  private static final int TF_OD_API_INPUT_SIZE = 300;
  private static final boolean TF_OD_API_IS_QUANTIZED = true;
  private static final String TF_OD_API_MODEL_FILE = "detect.tflite";
  private static final String TF_OD_API_LABELS_FILE = "file:///android_asset/pets_labels_list.txt";
`

Once updated, everything recompiles with the same output as before and it installs just fin on the device. TFL Classify and TFL Speech work the same as before, but the TFL Detect presents me with a "TFLite Demo has stopped." popup when ran. Using logcat, I was able to get the following stacktrace.

`
04-11 15:52:09.383 23876 23876 E tensorflow: DetectorActivity: Exception initializing classifier!

...

04-11 15:52:09.712 23876 23893 E AndroidRuntime: FATAL EXCEPTION: inference
04-11 15:52:09.712 23876 23893 E AndroidRuntime: Process: org.tensorflow.lite.demo, PID: 23876
04-11 15:52:09.712 23876 23893 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.List org.tensorflow.demo.Classifier.recognizeImage(android.graphics.Bitmap)' on a null object reference
04-11 15:52:09.712 23876 23893 E AndroidRuntime:        at org.tensorflow.demo.DetectorActivity$3.run(DetectorActivity.java:247)
04-11 15:52:09.712 23876 23893 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:789)
04-11 15:52:09.712 23876 23893 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:98)
04-11 15:52:09.712 23876 23893 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
04-11 15:52:09.712 23876 23893 E AndroidRuntime:        at android.os.HandlerThread.run(HandlerThread.java:65)
`

With the new error, I put the original files back and undid my changes to the source and recompiled a third time with every back to the way it was. I then installed that APK and the Detect demo was working again.

I am at a loss on what to try next as exchanging the model with the other provided sample assets should work? I did try a custom trained model of my own and got the same stacktrace with my own model. I can provide info on those models if it help at all.

Thanks in advance!

**Comment by Dayananda-V** [comment]
@blakefox  please use the proper labels file name in declaration will resolve your problem

  private static final String TF_OD_API_LABELS_FILE = "file:///android_asset/pets_labels_list.txt";
  
to

  private static final String TF_OD_API_LABELS_FILE = "pets_labels_list.txt";

**Comment by achandraa** [comment]
@blakefox - Were you able to resolve this issue ?

**Comment by blakefox** [comment]
Sorry I was out yesterday. I have not been able to resolve this issue as of  yet. I will try the fix Dayananda recommended above and report back shortly.

**Comment by blakefox** [comment]
A quick update,

Dayananda's fix did resolve the original issue of this ticket. Thank you very much for the advice.

Sadly the same crash now happens right as soon and the image feed starts. That being said, I think this new crash is due to my training data. If its not, I will report back with the new stack trace shortly.

**Comment by achandraa** [comment]
@Dayananda-V : Thank you for the support
@blakefox : That's great. For now I am closing the issue since it is resolved. Thanks!
